launch_testing_ros/launch_testing_ros/wait_for_topics.py

```diff
@@ -0,0 +1,110 @@
+# Copyright 2021 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from threading import Event
+from threading import Lock
+from threading import Thread
+
+import rclpy
+from rclpy.executors import SingleThreadedExecutor
+from rclpy.node import Node
+
+
+class WaitForTopics:
+    """Wait to receive messages on supplied nodes."""
+
+    def __init__(self, topic_tuples, timeout=5.0):
+        self.__ros_context = None
+        self.__ros_node = None
+        self.__ros_executor = None
+        self.topic_tuples = topic_tuples
+        self.timeout = timeout
+        self.start()
+
+    def start(self):
+        self.__ros_context = rclpy.Context()
+        rclpy.init(context=self.__ros_context)
+        self.__ros_executor = SingleThreadedExecutor(context=self.__ros_context)
+        self.__ros_node = MakeTestNode(name='test_node', node_context=self.__ros_context)
```

Comment:
It's possible we may want to create multiple instances of class in a test, so I think we should ensure the node name is unique by adding some random content to the string. It's also probably nice to hide the node from tooling by prefixing it with an underscore.

Reply:
Added underscore prefix and random characters at the end.